BPA/Ch.12/binmix_cov.stan

```diff
@@ -0,0 +1,58 @@
+// Binomial mixture model with covariates
+data {
+  int<lower=0> R;       // Number of sites
+  int<lower=0> T;       // Number of temporal replications
+  int<lower=0> y[R, T]; // Counts
+  vector[R] X;          // Covariate
+  int<lower=0> K;       // Upper bound of population size
+}
+
+transformed data {
+  int<lower=0> max_y[R];
+
+  for (i in 1:R)
+    max_y[i] <- max(y[i]);
+}
+
+parameters {
+  real alpha0;
+  real alpha1;
+  real beta0;
+  real beta1;
+}
+
+transformed parameters {
+  vector[R] log_lambda;
+  matrix[R, T] logit_p;
+
+  log_lambda <- alpha0 + alpha1 * X;
+  for (i in 1:R)
+    for (j in 1:T)
+      logit_p[i, j] <- beta0 + beta1 * X[i];
+}
```

Comment:
If you turn X into a real valued-vector, then you can do a lot of this with vector arithmtic more efficiently.